### PR TITLE
Accept pymongo 4.x for Python SDK

### PR DESCRIPTION
**Pull request by Abacn**
Test #25783

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Abacn** [comment]
Run Python MongoDBIO_IT

**Comment by Abacn** [comment]
Run Python MongoDBIO Load Test

File: sdks/python/setup.py [hunk 1]
```diff
@@ -250,7 +250,7 @@ def get_portability_package_data():
         # Use a strict upper bound.
         'numpy>=1.14.3,<1.25.0',   # Update build-requirements.txt as well.
         'objsize>=0.6.1,<0.7.0',
-        'pymongo>=3.8.0,<4.0.0',
+        'pymongo>=3.8.0,<5.0.0',
```
Comment (tvalentyn) [inline sdks/python/setup.py:253]: consider temporarily increasing the lower bound to 4.0.0 or above to make IT signal more reliable.

**Comment by Abacn** [comment]
From the log of both integration test and load test I see pymongo 4.3.3 is installed: https://ci-beam.apache.org/job/beam_PostCommit_Python_MongoDBIO_IT_PR/22/console https://ci-beam.apache.org/job/beam_python_mongoio_load_test/4666/consoleFull

Sure kick off again with narrower dep

**Comment by Abacn** [comment]
Run Python MongoDBIO_IT

**Comment by Abacn** [comment]
Run Python MongoDBIO Load Test

**Comment by Abacn** [comment]
Run Python MongoDBIO_IT

**Comment by Abacn** [comment]
Run Python MongoDBIO Load Test

**Comment by Abacn** [comment]
tests succeeded again with both SDK and container pymongo version change:
- https://ci-beam.apache.org/job/beam_PostCommit_Python_MongoDBIO_IT_PR/24/
- https://ci-beam.apache.org/job/beam_python_mongoio_load_test/4670/

reverting the lower bound change and mark as ready-for-review

**Comment by Abacn** [comment]
Run Python_Runners PreCommit

**Review by tvalentyn — APPROVED** [review]

**Pull request merged**